tests/inline_formsets/tests.py

```diff
@@ -133,6 +133,17 @@ def test_fk_name_not_foreign_key_field_from_child(self):
         with self.assertRaisesMessage(ValueError, msg):
             inlineformset_factory(Parent, Child, fk_name='school')
 
+
+    def test_foreign_key_on_proxy_model_with_fk_name(self):
+        # Regression test for #32975.
+        inlineformset_factory(SpecialJournalist, Article, fk_name='journalist', fields="__all__")
+
+
+    def test_foreign_key_on_proxy_model_without_fk_name(self):
+        # Regression test for #32975.
+        inlineformset_factory(SpecialJournalist, Article, fields="__all__")
+
+
```

Comment:
I would move regression tests to the `modeladmin.test_checks.FkNameCheckTests` with inline classes, e.g.
```python
    def test_proxy_model_parent(self):
        class Parent(Model):
            pass

        class ProxyChild(Parent):
            class Meta:
                proxy = True

        class ProxyProxyChild(ProxyChild):
            class Meta:      
                proxy = True 
                             
        class Related(Model):
            proxy_child = ForeignKey(ProxyChild, on_delete=CASCADE)
                                   
        class InlineFkName(admin.TabularInline):
            model = Related
            fk_name = 'proxy_child'
        
        class InlineNoFkName(admin.TabularInline):
            model = Related
        
        class ProxyProxyChildAdminFkName(admin.ModelAdmin):
            inlines = [InlineFkName, InlineNoFkName]
                                    
        self.assertIsValid(ProxyProxyChildAdminFkName, ProxyProxyChild)

```